src/problems/MetalAdvectionProblem/test_sne.cpp

```diff
@@ -0,0 +1,562 @@
+		amrex::Real phi_interp = (y1 + (y2 - y1) * (x_interp - x1) / (x2 - x1));
+		Phigas = FastMath::pow10(phi_interp);
+
+		double Phitot = Phist + Phidm + Phigas;
+
+		double rho, rho_disk, rho_halo;
+		rho_disk = rho01 * std::exp(-Phitot / std::pow(sigma1, 2.0));
+		rho_halo = rho02 * std::exp(-Phitot / std::pow(sigma2, 2.0)); // in g/cc
+		rho = (rho_disk + rho_halo);
+
+		double P = rho_disk * std::pow(sigma1, 2.0) + rho_halo * std::pow(sigma2, 2.0);
+
+		AMREX_ASSERT(!std::isnan(rho));
+
+		const auto gamma = HydroSystem<NewProblem>::gamma_;
+
+		state_cc(i, j, k, HydroSystem<NewProblem>::density_index) = rho;
+		state_cc(i, j, k, HydroSystem<NewProblem>::x1Momentum_index) = 0.0;
+		state_cc(i, j, k, HydroSystem<NewProblem>::x2Momentum_index) = 0.0;
+		state_cc(i, j, k, HydroSystem<NewProblem>::x3Momentum_index) = 0.0;
+		state_cc(i, j, k, HydroSystem<NewProblem>::internalEnergy_index) = P / (gamma - 1.);
+		state_cc(i, j, k, HydroSystem<NewProblem>::energy_index) = P / (gamma - 1.);
+		state_cc(i, j, k, Physics_Indices<NewProblem>::pscalarFirstIndex) = 1.e-5 / vol; // Injected tracer
+	});
+}
+
+void AddSupernova(amrex::MultiFab &mf, amrex::GpuArray<Real, AMREX_SPACEDIM> prob_lo, amrex::GpuArray<Real, AMREX_SPACEDIM> prob_hi,
+		  amrex::GpuArray<Real, AMREX_SPACEDIM> dx, SimulationData<NewProblem> const &userData, int level)
+{
+	// inject energy into cells with stochastic sampling
+	BL_PROFILE("QuokkaSimulation::Addsupernova()")
+
+	const Real cell_vol = AMREX_D_TERM(dx[0], *dx[1], *dx[2]); // cm^3
+	const Real rho_eint_blast = userData.E_blast / cell_vol;   // ergs cm^-3
+	const Real rho_blast = userData.M_ejecta / cell_vol;	   // g cm^-3
+	const int cum_sn = userData.SN_counter_cumulative;
+
+	for (amrex::MFIter iter(mf); iter.isValid(); ++iter) {
+		const amrex::Box &box = iter.validbox();
+		auto const &state = mf.array(iter);
+		auto const &px = userData.blast_x->table();
+		auto const &py = userData.blast_y->table();
+		auto const &pz = userData.blast_z->table();
+		const int np = userData.nblast;
+
+		amrex::ParallelFor(box, [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+			const Real xc = prob_lo[0] + static_cast<Real>(i) * dx[0] + 0.5 * dx[0];
+			const Real yc = prob_lo[1] + static_cast<Real>(j) * dx[1] + 0.5 * dx[1];
+			const Real zc = prob_lo[2] + static_cast<Real>(k) * dx[2] + 0.5 * dx[2];
+
+			for (int n = 0; n < np; ++n) {
+				Real x0 = NAN;
+				Real y0 = NAN;
+				Real z0 = NAN;
+				Real Rpds = 0.0;
+
+				x0 = std::abs(xc - px(n));
+				y0 = std::abs(yc - py(n));
+				z0 = std::abs(zc - pz(n));
+
+				if (x0 < 0.5 * dx[0] && y0 < 0.5 * dx[1] && z0 < 0.5 * dx[2]) {
+					state(i, j, k, HydroSystem<NewProblem>::density_index) += rho_blast;
+					state(i, j, k, HydroSystem<NewProblem>::energy_index) += rho_eint_blast;
+					state(i, j, k, HydroSystem<NewProblem>::internalEnergy_index) += rho_eint_blast;
+					state(i, j, k, Physics_Indices<NewProblem>::pscalarFirstIndex) += 1.e3 / cell_vol;
```

Comment:
For consistency, maybe `1.e3 / cell_vol` can be moved to a variable after `rho_blast` on line 208?